libyul/optimiser/ForLoopConditionOutOfBody.h

```diff
@@ -0,0 +1,56 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <libyul/optimiser/ASTWalker.h>
+#include <libyul/Dialect.h>
+
+namespace yul
+{
+
+/**
+ * Reverses the transformation of ForLoopConditionIntoBody.
+ *
+ * for { ... } 1 { ... } {
+ *   if iszero(c) { break }
+ *   ...
+ * }
+ *
+ * will be turned into
+ *
+ * for { ... } c { ... } {
+ *   ...
+ * }
+ *
```

Comment:
and ``for { ... } 1 { ... } { if c  { break } ... }`` will be turned into ``for { ... } iszero(c) { ... } { ... }``